snowfall/training/compute_embeddings.py

```diff
@@ -269,8 +270,17 @@ def compute_embeddings(
     low_scores = k2.index(scores, low.to(torch.int32))
     high_scores = k2.index(scores, high.to(torch.int32))
 
-    embedding_scores = low_scores * low_scale.unsqueeze(
-        -1) + high_scores * high_scale.unsqueeze(-1)
+    # It can happen that `high_scores` contain fake nnet_output,
+    # i.e., scores for arcs entering the final state are
+    # [0, -inf, -inf, ..., ]
+    #  embedding_scores = low_scores * low_scale.unsqueeze(
```

Comment:
I've disabled the interpolation here since `high_scores` may not come from
the `nnet_output`, i.e., it belongs to arcs entering the final state, which is
[0, -inf, -inf, ... -inf]`. Have not found an efficient approach to exclude these scores.